### ISSUE TITLE
gplay release outdated (3.29.2 instead of 3.30.3)

### ISSUE THREAD

**Issue by XueSheng-GIT** [nextcloud/android#13871]
Latest nextcloud gplay release is ~2~3.29.2 (https://play.google.com/store/apps/details?id=com.nextcloud.client):
![Image](https://github.com/user-attachments/assets/75825aa3-b040-4711-bdad-64efcf1b221b)

F-Droid offers version 3.30.0 and github indicates version 3.30.3 as latest.

Is it a bug that those newer releases are missing in gplay app store or is it done by intention?

**Comment by joshtrichards** [comment]
Agreed. As far as I can tell `gplay` seems stuck at v3.29.2. 

In the back of my mind I'd sort of noticed most bug reports that were for the `gplay` builds have remained v3.29.2 even up until today.

For lack of a better proxy, here are some stats:

- We haven't received any `gplay` specific crash bug reports  for 3.30.1 (in comparison to 15 for the `generic` build):
  - https://github.com/nextcloud/android/issues?q=is%3Aissue%20state%3Aopen%2030300190%20gplay
- And just two reports were received for 3.30.0 (in comparison to 70 for the `generic` build) [direct installs maybe?]:
  - https://github.com/nextcloud/android/issues?q=is%3Aissue%20state%3Aclosed%2030300090%20gplay

Cc: @AndyScherzinger, @ZetaTom, @tobiasKaminsky 

On the upside, that may mean #13738 impacted far fewer people.

**Comment by Markoise** [comment]
I have the same problem. I would use F-Droid as the source, but then the notifications no longer work, right?

**Comment by cmrtdev** [comment]
As of a yesterday the latest release is 3.30.4 (https://github.com/nextcloud/android/releases/tag/stable-3.30.4) and the Google Play Store still offers 3.29.2.

I'm actually looking forward to 3.30.x because of the sync improvements, any chance the release to Google Play Store can be tracked somewhere? I know it can take some time to happen, but it'd be nice to know that it's happening.

Thanks!

**Comment by tobiasKaminsky** [comment]
Google review is blocking us currently because we use the MANAGE_EXTERNAL_STORAGE permission, which was granted to us in early 2022, when it was introduced.
But suddenly they are not allowing us to use it.

Removing this permission would change auto upload behaviour quite a lot and thus I am in discussions with them, to keep this permission.

I will keep you updated.

**Comment by chrissi5120** [comment]
> Noticed this issue too today. Any ETA on this?

Are you guys just accepting this? I just filed a complaint with the european commission and the german Bundeskartellamt. 

This is screaming conflict of interest like nothing else.

**Comment by AndyScherzinger** [comment]
> Are you guys just accepting this?

We are appealing since July but keep facing rejections on the appeals

And yes, this is screaming conflict of interest like nothing else - I think so too

**Comment by MrRinkana** [comment]
Do they state the reason why the permission declaration gets denied?

Also this issue is technically closed since an update has shipped thanks to #14099, but it night be nice to track the status of the appeal here? Or should a dedicated issue be created?

**Comment by mieseprem** [comment]
I wonder why you aren't allowed to have the permission anymore.
Their documentation states thats your (our) usecase is okay: https://developer.android.com/training/data-storage/manage-all-files
(Hopefully this is the correct doc, I'm curious about the 'training' in the path)

**Comment by sol8712** [comment]
This is very HIGHLY anti-competitive if Google is allowing their own cloud apps this function but blocks other cloud providers from using the same functions in their store.

For our EU users this needs to be reported to the EU Commission for anti-trust, in the US its the FTC.

Now that their both agency's have been handing out fines like candy, now is the time to submit stuff like this for their investigations.

Unless this is Nextcloud refusing to budge or change....then thats a whole different issue......

**Comment by sol8712** [comment]
If Nextcloud is open source and the community is expected to assist with development, bugs, maintaining various apps, the community needs some transparency on what Googles reasons are for declining so developers can work within Googles rules.

If Googles own rules or decisions maybe breaking some laws or regulations people need that info to be able to report them to the relevant authority.

**Comment by Markoise** [comment]
Is there at least a way to receive push notifications when you install the app via F-Droid?

**Comment by salixor** [comment]
> > Couldn't a more tech-savvy user switch to the F-Droid release? Is this possible without losing data?
> 
> You can't, because the F-Droid release is signed with a different key than the Google Play version. You have to completely uninstall and reinstall the app. You could try to back up your data beforehand with `adb backup`, but YMMV because that command is deprecated. 
> 
> However! The APK in GitHub releases is signed with the same key as the Google Play version. So you could switch to that and use Obtanium to keep it up to date automatically.

Hey @strugee, just to clear it up for me and probably other users. Does the GitHub version still have the permission?

I've been using this one for a while, but the changelog seems to mean the permission is removed as well.

Your last paragraph however makes me think it is okay to install?

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky might be able to give some more details on Google's rejections but to my knowledge Google doesn't provide a justification at all, they just state that you violate their rules (in this case making use of the permission in general).

This permission is forbidden to be used since 2022 except you were able to make a case on why you still need this permission, which we did back then. In July 2024 Google changed their mind and rejected any app update to the store until we remove the permission. We made appeals to them various times July till December. And to be clear we also stated that they themselves as well as competitors still use this permission hence we should be allowed too did help but got the reply: "Just because others still have the permission doesn't make your case valid...". 

So after 4 months of no ability to publish an app updates to the store we gave in and removed the permission. So if anybody has any ideas or maybe contacts at Google to have a better way to get this address beyond reporting it to authorities, happy to pursue them.

**Comment by AndyScherzinger** [comment]
> I've been using this one for a while, but the changelog seems to mean the permission is removed as well.

@salixor yes, has also been removed on the GH version, else we would need to maintain and publish yet another build flavor which is why we decided against it, but are willing to reconsider in the light of the discussion while to me this is merely a workaround adding extra work but not solving the situation for GPlay users.

**Comment by enoch85** [comment]
Is it just me, or is this issue resolved already?

![Image](https://github.com/user-attachments/assets/57594cdf-ec87-4a88-a6bc-aa78cc432fd9)

**Comment by marki555** [comment]
Yes, the new version 3.30.6 is already available via GPlay, however there are 2 issues (both already have Issue opened here):

1) It doesn't automatically request the new Media permission, so auto-upload just stops to work (you need to go to Auto-upload settings and only then it will ask for the permission)

2) It doesn't have MEDIA_LOCATION permission, so all uploaded media files will be silently stripped off all GPS metadata by Android system.

**Comment by Markoise** [comment]
> 2) It doesn't have MEDIA_LOCATION permission, so all uploaded media files will be silently stripped off all GPS metadata by Android system.

That is a big problem.
Can it be fixed? Otherwise it makes the whole platform unusable for me

**Comment by bilogic** [comment]
Title says `gplay release outdated`, it seems to me that has been fixed, let's not digress further

**Comment by enoch85** [comment]
> Title says `gplay release outdated`, it seems to me that has been fixed, let's not digress further

Close this issue then?

cc @XueSheng-GIT

**Comment by bilogic** [comment]
> Close this issue then?

Yes, and I think it should be a unilateral decision by the maintainers:

1. There is no specific impact to begin with, only a question if the different versions on github and play store is intentional or a bug
2. And there is no clear answer if this is correct or incorrect behavior
3. But the version on play is now 3.30.7 and latest release on this repo is 3.30.7

**Comment by enoch85** [comment]
Closing based on @bilogic reasoning.

Feel free to reopen if the specific issue is unresolved.
